src/samples/audio/detect_test.c

```diff
@@ -48,6 +49,8 @@
 /* default number of samples before detection is activated  */
 #define KEYPHRASE_DEFAULT_PREAMBLE_LENGTH 0
 
+#define DEFAULT_DETECT	0
```

Comment:
Don't break Intel's CI. Put the value that would run Intel's default detect_test rather than our own here.

Reply:
I would actually go like this

```
       /* using default processing function */
»       cd->detect_func = default_detect_test;
#ifdef KWD_NN_DETECT_TEST
         cd->detect_func = kwd_nn_detect_test;
#endif                
```

Reply:
It should be:
#ifndef KWD_NN_DETECT_TEST
       cd->detect_func = default_detect_test;
#else 
         cd->detect_func = kwd_nn_detect_test;
#endif
Because default_detect_test will be defined only if KWD_NN_DETECT_TEST is not defined (compiler warning treated as error). If KWD_NN_DETECT_TEST is defined than we don't have default_detect_test defined and we cannot call it. A little bit complicated but not so much :)

Reply:
We shouldn't have multiple such variables that denote the exact same thing without a good reason. Hardcoding #defines, especially in a .c file (in a header file in src/platform/*/include it's probably fine, in a generic .c file it's code smell at the very least)

Reply:
@cristinailie the function `default_detect_test` is always defined because it is implemented in the same file we are using it. What i just want to do is to make the code very easy to be modified in the future when will support runtime selection of detect function.

Just right now we want to tell that if CONFIG_KWD_NN_DETECT_TEST is defined then just override `default_detect_test` with `kwd_nn_detect_test`. In this way the code is super simple and we avoid putting the reader to think to much.

Just look at this code:

```
       /* using default processing function */
»       cd->detect_func = default_detect_test;
#ifdef CONFIG_KWD_NN_DETECT_TEST
         cd->detect_func = kwd_nn_detect_test;
#endif  
```
If CONFIG_KWD_NN_DETECT_TEST the code will translate to:

```
cd->detect_func = default_detect_test;
cd->detect_func = kwd_nn_detect_test;
```

This will compile fine and will make kwd_nn_detect_test the detect_test function used by KWD.

If CONFIG_NN_DETECT_TEST is not defined then the code stays as it originally was.